tpp-app/src/services/utils/helpers.js

```diff
@@ -16,3 +17,44 @@ const initializeEmptyYear = (yearNumber) => {
 
   return year
 }
+
+/**
+ * Check if the date, month, year combination is a valid date.
+ * @param day number
+ * @param month number (January = 1)
+ * @param year number
+ * @return True or False boolean
+ */
+export const isValidDate = (day, month, year) => {
+    // Check the ranges of month and year
+    if (year < 1000 || year > 3000 || month == 0 || month > 12)
+        return false;
+
+    let monthLength = [ 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 ];
+
+    // Adjust for leap years
+    if (year % 400 == 0 || (year % 100 != 0 && year % 4 == 0))
+        monthLength[1] = 29;
+
+    // Check the range of the day
+    return day > 0 && day <= monthLength[month - 1];
+};
+
+/**
+ * Retrieves the user's symptom data for the given date.
+ * @param day number
+ * @param month number (January = 1)
+ * @param year number
+ */
+export const GETsymptomsForDate = async (day, month, year) => {
+  if (isValidDate(day, month, year)) {
+    // Get the year's data (value could be null if year is empty)
+    const yearData = JSON.parse(await AsyncStorage.getItem(year.toString()));
+
+    // Return symptoms for that day or empty symptoms object if it doesn't exist
+    return yearData[month-1][day-1] ? new Symptoms(yearData[month-1][day-1]) : new Symptoms();
```

Comment:
doesn't this create a `Symptoms` object with `flow = yearData[month-1][day -1]` and all other arguments null?

Reply:
bruh you right, I'm dumb